### DISCUSSION TITLE
Custom Style Definition - ETA?

### DISCUSSION THREAD

**Discussion by diehl** [protomaps/PMTiles#392] [General]
Any estimate on when there will be a workflow in place for defining and utilizing custom styles for PMTiles basemaps? I'd love to replace my existing Mapbox basemap that I've designed in Mapbox Studio with a PMTiles analog.

**Comment by bdon** [comment]
What degree of customization you need? We see the primary point of customization being defining a custom color scheme like in https://github.com/protomaps/basemaps/pull/234

Maybe this should be transferred to https://github.com/protomaps/basemaps/discussions ?

**Reply by diehl** [reply]
My needs are rather modest actually, as my basemap is mainly territorial boundaries and place names. But I would like to have control over the colors and text label fonts. 

After reading [this article](https://www.kschaul.com/post/2023/02/16/how-the-post-is-replacing-mapbox-with-open-source-solutions/) some time back, I was left with the impression that one could use Maputnik to define any arbitrary style and realize it with PMTiles. Is there something misleading about this post?

Happy to move this discussion elsewhere if that would be better!

**Reply by bdon** [reply]
If you need to control the colors and text label fonts you can do that with the MapLibre JSON styling language, which will refer to features in the specific tileset you're using.

**Reply by diehl** [reply]
Sounds great. What I'm a bit unclear on are the specifics behind the process laid out in the post above. Is the idea that one first designs a style in Maputnik and makes note of the OSM layers utilized in the resulting stylized map. Then one builds MBTiles with the requisite OSM layers. And then finally one converts the MBTiles to PMTiles to visualize with MapLibre? 

I'm coming to all this from the world of Python visualization, having hit some fundamental limitations I need to get round. So I'm trying to wrap my head around the details. Appreciate any pointers!

**Reply by bdon** [reply]
* maputnik is an editor for MapLibre JSON styles
* the JSON style needs to reference layers and tags that exist in your tileset
* your tileset can be built from OSM, whatever data. pmtiles is a storage format that can be decoded on the client, the format is orthogonal to the tile content you are delivering. right now if you want to use maputnik you need to reference a tileset Z/X/Y url which you can accomplish with `pmtiles serve` or cloudflare workers

**Reply by diehl** [reply]
Much appreciated @bdon - thank you!

**Reply by bdon** [reply]
the pmtiles viewer at https://pmtiles.io just assigns a random-ish color to each layer. If you mean the https://maps.protomaps.com basemaps viewer, click "Get Style JSON"

**Reply by diehl** [reply]
Thanks @bdon! I just discovered it and deleted my question a moment before you answered. ;-) Much appreciated!
